src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -1791,11 +1790,12 @@ Crypto::SymmetricKeyContext * GroupDataProviderImpl::GetKeyContext(FabricIndex f
             // Group found, get the keyset
             KeySetData keyset;
             VerifyOrReturnError(keyset.Find(mStorage, fabric, mapping.keyset_id), nullptr);
-            OperationalKey * key = keyset.GetCurrentKey();
-            if (nullptr != key)
+            GroupOperationalCredentials * creds = keyset.GetCurrentGroupCredentials();
+            if (nullptr != creds)
             {
-                return mKeyContexPool.CreateObject(*this, ByteSpan(key->value, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES),
-                                                   key->hash);
+                return mGroupKeyContexPool.CreateObject(
+                    *this, ByteSpan(creds->encryption_key, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES), creds->hash,
+                    ByteSpan(creds->privacy_key, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES));
```

Comment:
I don't see how `privacy_key` can possibly be valid here.  Where are we setting it?  I can't find it.

Reply:
Should be valid now that privacy key derivation is in Deserialize.